packages/reactDom/src/ReactServerDOM.ml

```diff
@@ -654,22 +663,25 @@ let rec to_html ~(fiber : Fiber.t) (element : React.element) : (Html.element * j
         context.push (error_chunk_script index error_json);
         context.push (chunk_html_script index Html.null);
         Lwt.return (html, Model.suspense_placeholder ~key ~fallback:model_fallback index))
-  | Provider children -> to_html ~fiber children
-  | Consumer children -> to_html ~fiber children
+  | Provider children -> to_html ~fiber children ~withBodyHtml
+  | Consumer children -> to_html ~fiber children ~withBodyHtml
   (* TODO: There's a task to remove InnerHtml in ReactDOM and use Html.raw directly. Here is still unclear what do to since we assing dangerouslySetInnerHTML to the right prop on the model. Also, should this model be `Null? *)
   | InnerHtml innerHtml -> Lwt.return (Html.raw innerHtml, `Null)
 
-and elements_to_html ~fiber elements =
-  let%lwt html_and_models = elements |> Lwt_list.map_p (to_html ~fiber) in
+and elements_to_html ~fiber elements ~withBodyHtml =
+  let%lwt html_and_models = elements |> Lwt_list.map_p (to_html ~fiber ~withBodyHtml) in
   (* TODO: List.split is not tail recursive *)
   let htmls, model = List.split html_and_models in
   Lwt.return (Html.list htmls, `List model)
 
-let is_root_html_node element =
+let rec is_root_html_node element =
   match (element : React.element) with
-  | Lower_case_element { tag; _ } -> tag = "html"
-  | React.Fragment (React.List [ Lower_case_element { tag = "html"; _ }; _ ]) -> true
-  | _ -> false
+  | Lower_case_element { tag = "html"; _ } -> Lwt.return true
+  | React.Fragment (React.List [ Lower_case_element { tag = "html"; _ }; _ ]) -> Lwt.return true
+  | Async_component (_, component) ->
+      let%lwt element = component () in
+      is_root_html_node element
```

Comment:
@davesnx turns out we weren't handling the Async html components, so for this line: https://github.com/ml-in-barcelona/server-reason-react/pull/266/files#diff-c42b16a38191961579f8f6e766741e8fee7697090c98a4cec9a890e6fedfcc71R91-R93 we would fall as `not(is_root_html_node)`

Reply:
I changed the RouterRSCNoSSR not to be an async component. But the is_root_html_node was still failing, I identified that we weren't covering this case: https://github.com/ml-in-barcelona/server-reason-react/blob/434574dc1c6d134243988eead38c1dcde63948dd/packages/reactDom/src/ReactServerDOM.ml#L680